packages/node/src/policy-gen/policy-gen-compartment-class.js

```diff
@@ -224,21 +249,27 @@ const makeGetOverriddenResourceNames = (compartmentMap, resources) => {
  * to hand-craft a giant policy override.
  *
  * @param {CompartmentMapDescriptor} compartmentMap
- * @param {LavaMoatPolicyOverrides} [policyOverride]
+ * @param {LavaMoatPolicy} [policyOverride]
  * @returns {typeof Compartment} Either the original `Compartment` or a
  *   `PolicyGenCompartment`
  * @internal
  */
 export const makePolicyGenCompartment = (compartmentMap, policyOverride) => {
-  if (!policyOverride || !policyOverride.resources) {
+  if (!policyOverride?.resources) {
     return Compartment
   }
 
-  // only create this function and resulting `Compartment` if we have resources
-  // in the policy override
+  const compartmentsByLabel = fromEntries(
+    entries(compartmentMap.compartments).map(([, compartment]) => [
+      compartment.label,
+      compartment,
+    ])
+  )
+
   const getOverriddenResourceNames = makeGetOverriddenResourceNames(
     compartmentMap,
-    policyOverride.resources
+    compartmentsByLabel,
```

Comment:
No need to change the signature of `makeGetOverriddenResourceNames` because `compartmentsByLabel` can be created on the inside.